components/TaskInputForm/index.js

```diff
@@ -114,6 +114,33 @@ export default function Form({ onSubmit, formName, defaultData }) {
     setAddingSubtask(false);
   }, [addingSubtask]);
 
+  useEffect(() => {
+    if (defaultData?.title) {
+      titleInputRef.current.value = defaultData.title;
+    }
+    if (defaultData?.subtasks) {
+      const defaultSubtasks = defaultData.subtasks.map((subtask) => {
+        return { id: uuidv4(), value: subtask };
+      });
+      setSubtasks(defaultSubtasks);
+    }
+    if (defaultData?.tags) {
+      const defaultTags = defaultData.tags.map((tag, index) => {
+        return { id: String(index + 1), text: tag };
+      });
+      setTags(defaultTags);
+    }
+    if (defaultData?.priority) {
+      setSelectedPrio(defaultData.priority);
+    }
+    if (defaultData?.deadline) {
+      const formattedDefaultDeadline = new Date(defaultData.deadline)
+        .toISOString()
+        .split("T")[0];
+      document.getElementById("deadline").value = formattedDefaultDeadline;
+    }
+  }, [defaultData]);
```

Comment:
I think that you do not need to work with useEffect here. You tried to solve it with defaultValue first and this should be the preferred and and less complicated way to do this. See for reference how we implemented defaultValues in our recap project 6: tourio app:

```jsx
export default function Form({ onSubmit, formName, defaultData }) {
  function handleSubmit(event) {
    event.preventDefault();
    const formData = new FormData(event.target);
    const data = Object.fromEntries(formData);
    onSubmit(data);
  }

  return (
    <FormContainer aria-labelledby={formName} onSubmit={handleSubmit}>
      <Label htmlFor="name">Name</Label>
      <Input
        id="name"
        name="name"
        type="text"
        defaultValue={defaultData?.name}
      />
      <Label htmlFor="image-url">Image Url</Label>
      <Input
        id="image-url"
        name="image"
        type="text"
        defaultValue={defaultData?.image}
      />
      <Label htmlFor="location">Location</Label>
      <Input
        id="location"
        name="location"
        type="text"
        defaultValue={defaultData?.location}
      />
      <Label htmlFor="map-url">Map Url</Label>
      <Input
        id="map-url"
        name="mapURL"
        type="text"
        defaultValue={defaultData?.mapURL}
      />
      <Label htmlFor="description">Description</Label>
      <Textarea
        name="description"
        id="description"
        cols="30"
        rows="10"
        defaultValue={defaultData?.description}
      ></Textarea>
      <StyledButton type="submit">
        {defaultData ? "Update place" : "Add place"}
      </StyledButton>
    </FormContainer>
  );
}
```

And the [React docs have a good article](https://react.dev/learn/you-might-not-need-an-effect) about when you do not need an useEffect.

Reply:
I will try again. I was just lost when it came to priority and tags and so on.